validator/sawtooth_validator/gossip/signature_verifier.py

```diff
@@ -78,25 +76,24 @@ def validate_batch(batch):
     if not valid:
         LOGGER.debug("batch failed signature validation: %s",
                      batch.header_signature)
+        return False
 
     # validate all transactions in batch
-    total = len(batch.transactions)
-    index = 0
-    while valid and index < total:
-        txn = batch.transactions[index]
+    for txn in batch.transactions:
         valid = validate_transaction(txn)
-        if valid:
-            txn_header = TransactionHeader()
-            txn_header.ParseFromString(txn.header)
-            if txn_header.batcher_pubkey != header.signer_pubkey:
-                LOGGER.debug("txn batcher pubkey does not match signer"
-                             "pubkey for batch: %s txn: %s",
-                             batch.header_signature,
-                             txn.header_signature)
-                valid = False
-        index += 1
-
-    return valid
+        if not valid:
+            return False
```

Comment:
You can, _I think_, also replace the `for` loop with:

```
if not all(map(validate_transaction, batch.transactions)):
    return False
```

Since `map` returns an iterable and `all` is supposed to short-circuit, you get the same result.

Same goes for `validate_batches`